DECLARING_DEPENDENCIES.md

```diff
@@ -0,0 +1,210 @@
+you need, because the transitive dependencies of that library should
+all be compatible as well.
+
+However, if your project independently pulls in transitive
+dependencies at different versions, you may also need to specify
+lower-level BOMs to ensure compatibility, since each BOM only controls
+the versions of the library that generates it. Since the GCP Java
+libraries follow semver, you should be able to pick the highest
+version in your dependency tree for any particular library. Here is
+the process to follow for each library that has BOM support:
+
+1. Identify the highest version seen for any artifact produced by that library
+  a. For example, if you see `gax:1.34.0` and `gax-grpc:1.42.0`, then use version 1.42.0
+2. Import the BOM corresponding to that library using the highest version you see
+  a. For the example above, use `gax-bom:1.42.0`
+3. Repeat as long as there are other libraries with BOM support where
+   you see multiple versions of the library's artifacts
+
+Note that there will not necessarily be a BOM available at the version
+you need. See the BOM reference section below for the first
+available version of each BOM.
+
+## BOM reference
+
+None of the GCP open source Java libraries were initially released
+with BOMs - they were added later to provide better consistency for
+users. The following table shows the first version that each library
+published a BOM for, and the BOM artifact name.
+
+| library | BOM artifact | First available version |
+| --- | --- | --- |
+| guava | com.google.guava:guava-bom | 27.1-jre/27.1-android (08-Mar-2019) |
+| protobuf | com.google.protobuf:protobuf-bom | 3.7.0 (06-Mar-2019) |
+| grpc-java | io.grpc:grpc-bom | 1.19.0 (27-Feb-2019) |
+| gax-java | com.google.api:gax-bom | 1.34.0 (19-Oct-2018) |
+| google-http-java-client | com.google.http-client:google-http-client-bom | 1.27.0 (09-Nov-2018) |
+| google-oauth-java-client | com.google.oauth-client:google-oauth-client-bom | 1.27.0 (10-Nov-2018) |
+| google-api-java-client | com.google.api-client:google-api-client-bom | 1.27.0 (12-Nov-2018) |
+| google-cloud-java | com.google.cloud:google-cloud-bom | 0.32.0-alpha (11-Dec-2017)  |
+| beam | org.apache.beam:beam-sdks-java-bom | 2.10.0 (06-Feb-2019) |
+
+## Intrinsic conflicts
+
+It is possible for GCP open source Java libraries to have conflicts
+with each other that cannot be resolved when the user follows the
+recommendations of this document. Such conflicts are called intrinsic
+conflicts. There is an ongoing effort to ensure that intrinsic
+conflicts are avoided among Google open source Java libraries. A
+dashboard that reports the current results of compatibility checks is
+accessible from the [Cloud Open Source Java Dashboard](https://storage.googleapis.com/cloud-opensource-java-dashboard/dashboard/target/dashboard/dashboard.html).
+As of the time of this writing, there are still some conflicts that
+are in the process of being fixed, but they should not be encountered
```

Comment:
there are still some conflicts that
are in the --> some conflicts are still in the